designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/table/ui/TableDimensions.kt

```diff
@@ -0,0 +1,274 @@
+    val tableEndExtraScroll: Dp = Spacing.Spacing6,
+) {
+
+    private var currentExtraSize: MutableMap<String, Int> = mutableMapOf()
+    private fun extraWidthInTable(tableId: String): Int = extraWidths[tableId] ?: 0
+
+    var textInputHeight = 0
+
+    /**
+     * Update the width of the row header.
+     * @param tableId The ID of the table.
+     * @return The updated dimensions.
+     */
+    fun rowHeaderWidth(tableId: String): Int {
+        return (rowHeaderWidths[tableId] ?: defaultRowHeaderWidth) + extraWidthInTable(tableId)
+    }
+
+    internal fun defaultCellWidthWithExtraSize(
+        tableId: String,
+        totalColumns: Int,
+        hasExtra: Boolean = false,
+    ): Int = defaultCellWidth +
+        extraSize(tableId, totalColumns, hasExtra) +
+        extraWidthInTable(tableId)
+
+    internal fun columnWidthWithTableExtra(tableId: String, column: Int? = null): Int =
+        (columnWidth[tableId]?.get(column) ?: defaultCellWidth) + extraWidthInTable(tableId)
+
+    internal fun headerCellWidth(
+        tableId: String,
+        column: Int,
+        headerRowColumns: Int,
+        totalColumns: Int,
+        hasTotal: Boolean = false,
+    ): Int {
+        val rowHeaderRatio = totalColumns / headerRowColumns
+
+        val result = when {
+            rowHeaderRatio != 1 -> {
+                val maxColumn = rowHeaderRatio * (1 + column) - 1
+                val minColumn = rowHeaderRatio * column
+                (minColumn..maxColumn).sumOf {
+                    columnWidthWithTableExtra(tableId, it) +
+                        extraSize(tableId, totalColumns, hasTotal, column)
+                }
+            }
+            else -> columnWidthWithTableExtra(tableId, column) +
+                extraSize(tableId, totalColumns, hasTotal, column)
+        }
+        return result
+    }
+
+    internal fun extraSize(tableId: String, totalColumns: Int, hasTotal: Boolean, column: Int? = null): Int {
+        val screenWidth = totalWidth
+        val tableWidth = tableWidth(tableId, totalColumns, hasTotal)
+        val columnHasResizedValue = column?.let {
+            columnWidth[tableId]?.containsKey(it)
+        }
+
+        return if (tableWidth < screenWidth && columnHasResizedValue != true) {
+            val totalColumnCount = 1.takeIf { hasTotal } ?: 0
+            val columnsCount = totalColumns + totalColumnCount
+            ((screenWidth - tableWidth) / columnsCount).also {
+                currentExtraSize[tableId] = it
+            }
+        } else {
+            0
+        }
+    }
+
+    private fun tableWidth(tableId: String, totalColumns: Int, hasTotal: Boolean): Int {
+        val totalCellWidth = defaultCellWidth.takeIf { hasTotal } ?: 0
+        return rowHeaderWidth(tableId) + defaultCellWidth * totalColumns + totalCellWidth
+    }
+
+    private fun updateAllWidthBy(tableId: String, widthOffset: Float): TableDimensions {
+        val newWidth = (extraWidths[tableId] ?: 0) + widthOffset - 11
```

Comment:
What does 11 represent? why not 12 or 10?

Reply:
I'm not sure why, maybe is the sum of some padding or margins. @Balcan , @mmmateos , do you know why?